lib/modules/platform/github/index.ts

```diff
@@ -739,8 +739,10 @@ export async function findPr({
 const REOPEN_THRESHOLD_MILLIS = 1000 * 60 * 60 * 24 * 7;
 
 async function ensureBranchSha(branchName: string, sha: string): Promise<void> {
-  const refUrl = `/repos/${config.repository}/git/refs/heads/${branchName}`;
+  const commitUrl = `/repos/${config.repository}/git/commits/${sha}`;
+  await githubApi.head(commitUrl, { memCache: false });
```

Comment:
this will now throw when 404. what's the desired behavior? we need to push the Sha again?

Reply:
I think if we are trying to push a SHA which doesn't exist then we have no alternative but to throw - and I think this should really be an _error_ because it seems like a problem in our logic somehow. We should maybe try/catch/log/throw